plugins/woocommerce/includes/class-wc-post-data.php

```diff
@@ -621,6 +635,376 @@ public static function set_object_terms( $object_id, $terms, $tt_ids, $taxonomy,
 			self::delete_product_query_transients();
 		}
 	}
+
+	/**
+	 * Regenerates attribute summaries for a list of variations.
+	 *
+	 * @since 10.2.0
+	 * @param array $variation_ids Array of variation IDs.
+	 */
+	public static function regenerate_variation_summaries( $variation_ids ) {
+		if ( empty( $variation_ids ) ) {
+			return;
+		}
+
+		$variation_ids = array_unique( array_filter( array_map( 'intval', $variation_ids ) ) );
+
+		foreach ( $variation_ids as $variation_id ) {
+			self::regenerate_variation_attribute_summary( $variation_id );
+		}
+	}
+
+	/**
+	 * Regenerates the attribute summary for a single variation.
+	 *
+	 * @since 10.2.0
+	 * @param int $variation_id Variation ID.
+	 */
+	public static function regenerate_variation_attribute_summary( $variation_id ) {
+		global $wpdb;
+
+		$product = wc_get_product( $variation_id );
+		if ( ! $product || ! $product->is_type( 'variation' ) ) {
+			return;
+		}
+
+		$data_store = WC_Data_Store::load( 'product-variation' );
+		if ( $data_store->has_callable( 'get_attribute_summary' ) ) {
+			$new_summary     = $data_store->get_attribute_summary( $product );
+			$current_excerpt = get_post_field( 'post_excerpt', $variation_id );
+			if ( $new_summary === $current_excerpt ) {
+				return;
+			}
+
+			/**
+			* Update directly via $wpdb for performance: Avoid firing save_post hooks, loading full post objects,
+			* and creating revisions. This is safe here as we're only updating post_excerpt.
+			*/
+			$wpdb->update(
+				$wpdb->posts,
+				array( 'post_excerpt' => $new_summary ),
+				array( 'ID' => $variation_id )
+			);
+			clean_post_cache( $variation_id );
+			/**
+			* Fires after the attribute summary of a product variation has been updated.
+			*
+			* @since 10.2.0
+			* @param int $variation_id The ID of the product variation.
+			*/
+			do_action( 'woocommerce_updated_product_attribute_summary', $variation_id );
+		}
+	}
+
+	/**
+	 * Gets the threshold for synchronous regeneration of variation summaries.
+	 *
+	 * @since 10.2.0
+	 * @return int
+	 */
+	public static function get_regen_threshold() {
```

Comment:
Given there's other code in this class that's unrelated to variations, maybe it'd be good to rename this function in a way that better conveys it only affects regeneration of variation-related info.